spec/support/contexts/batch_delete.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+shared_context "batch delete" do
+
+  shared_examples_for "select all on page" do |options|
+    scenario do
+      visit("/en/m/#{mission.compact_name}/#{options[:klass]}")
+      search_for(options[:query]) unless options[:query].nil?
+      click_on("Select All")
+      click_on(options[:link])
+      confirm_delete_msg = options[:num] == 1 ? "Are you sure you want to delete this #{options[:klass].singularize}?" :
+        "Are you sure you want to delete these #{options[:num]} #{options[:klass]}?"
+      expect(accept_alert).to eq(confirm_delete_msg)
+      num = options[:query].nil? && options[:klass] == "users" ? options[:num] - 1 : options[:num]
```

Comment:
Need a comment on this weird edge case. Also it's confusing to have a `num` variable along with `options[:num]`. Can you just update `options[:num]`?

Reply:
done